java/src/main/resources/com/dke/data/agrirouter/proto/technicalmessagetype/gps.proto

```diff
@@ -36,8 +36,18 @@ message GPSList{
         uint32 number_of_satellites = 7;
         google.protobuf.Timestamp gps_utc_timestamp = 8;
         FieldStatus field_status = 9;//Are we on a road, in a field or just somewhere offroad?
+        uint32 source_device_index = 10;//The index of the source device in the SourceDeviceList
         repeated google.protobuf.Any extension = 2048;  // Used for proprietary purposes
     }
+
+    message SourceDevice{
+        uint32 index = 1;//The index within the list of SourceDevices; Not necessarilly all values are there
+        string display_name = 2;//Proprietary name for displaying (might change during operation)
+        string agrirouter_id = 3;//The endpoint_id as mentioned in the ListEndpoints Result
+        string internal_id = 4;//A proprietary but static id for sources outside of agrirouter
```

Comment:
This could be misleading. This is an ID for sources outside the agrirouter, but is named as `internalId`? Perhaps there is a better name for that?